src/Views/Wallpaper.vala

```diff
@@ -479,8 +412,12 @@ public class PantheonShell.Wallpaper : Gtk.Grid {
                     return;
                 }
 
+                if (WallpaperOperation.get_is_file_in_bg_dir (file)) {
+                    return;
```

Comment:
Returning here will omit the Gtk.drag_finish call.  Not sure whether you want to signal success or failure at this point.

Reply:
> Not sure whether you want to signal success or failure at this point.

Maybe we can signal success because just the wallpaper is not copied to bg dir and no errors occurred in this condition.

Reply:
@jeremypw Thank you for your review, fixed!